packages/hooks/src/useOauth2.ts

```diff
@@ -39,9 +44,60 @@ const useOauth2 = ({ handleLogout }: { handleLogout: () => Promise<void> }) => {
     };
 
     const logoutHandler = async () => {
+        if (process.env.REMOTE_CONFIG_URL) {
+            const flags = await fetch(process.env.REMOTE_CONFIG_URL)
+                .then(res => res.json())
+                .catch(() => FIREBASE_INIT_DATA);
+            if (process.env.RUDDERSTACK_KEY && flags?.tracking_rudderstack) {
+                const utm_cookie = Cookies.get('utm_data');
+                const ppc_campaign_cookies =
+                    utm_cookie === 'null' || !utm_cookie
+                        ? {
+                              utm_source: 'no source',
+                              utm_medium: 'no medium',
+                              utm_campaign: 'no campaign',
+                              utm_content: 'no content',
+                          }
+                        : JSON.parse(utm_cookie);
+
+                const analytics_config_config = {
+                    loggedIn: false,
+                    account_type: 'unlogged',
+                    app_id: String(getAppId()),
+                    device_language: navigator?.language || 'en-EN',
+                    user_language: getLanguage().toLowerCase(),
+                    country: undefined,
+                    utm_source: ppc_campaign_cookies?.utm_source,
+                    utm_medium: ppc_campaign_cookies?.utm_medium,
+                    utm_campaign: ppc_campaign_cookies?.utm_campaign,
+                    utm_content: ppc_campaign_cookies?.utm_content,
+                    domain: window.location.hostname,
+                    url: window.location.href,
+                    network_type: navigator.connection?.effectiveType,
+                    network_rtt: navigator.connection?.rtt,
+                    network_downlink: navigator.connection?.downlink,
+                    residence_country: 'undefined',
+                };
+                const config = {
+                    growthbookKey: flags.marketing_growthbook ? process.env.GROWTHBOOK_CLIENT_KEY : undefined,
+                    growthbookDecryptionKey: flags.marketing_growthbook
+                        ? process.env.GROWTHBOOK_DECRYPTION_KEY
+                        : undefined,
+                    rudderstackKey: process.env.RUDDERSTACK_KEY,
+                    growthbookOptions: {
+                        attributes: {
+                            ...analytics_config_config,
+                        },
+                    },
+                };
+                Analytics.setAttributes(analytics_config_config);
+                await Analytics?.initialise(config);
```

Comment:
[Q]: why do we need to initialize the whole Analytics package again? 🤔 only updating the attributes would be enough.

Reply:
It will work only when we re-initialize it for the logout session.can you check it and let me know if it is possible in any other way.

Reply:
I'm looking into this. will update you on this